src/Pim/Bundle/CatalogBundle/Doctrine/Common/Saver/ProductSaver.php

```diff
@@ -115,4 +116,49 @@ protected function validateProduct($product)
             );
         }
     }
+
+    /**
+     * Drops the current completenesses from the database for a list of products.
+     *
+     * @param ProductInterface[] $products
+     */
+    protected function dropCompletenesses(array $products)
+    {
+        $completenessIDs = [];
+        foreach ($products as $product) {
+            $completenesses = $product->getCompletenesses();
```

Comment:
It could be done at the same time than extracting it in a dedicated remover, as Samir suggested.